packages/gatsby/src/schema/node-model.js

```diff
@@ -210,11 +210,12 @@ class LocalNodeModel {
       result.forEach(node => this.trackInlineObjectsInRootNode(node))
     }
 
-    if (pageDependencies) {
-      return this.trackPageDependencies(result, pageDependencies)
-    } else {
-      return result
+    if (typeof pageDependencies.connectionType === `undefined`) {
```

Comment:
NIT: despite setting default value I think we can crash with unhelpful message if someone did `getAllNodes(<args>, null)` - so maybe just let's make  sure we check for `pageDependencies` being defined before trying to use `pageDependencies.connectionType`?

Reply:
Hmmm, actually above is probably incorrect because `ContextualNodeModel` always creates something with `_getFullDependencies`? and then this implies that our previous check for `pageDependencies` existance here is/was not needed? not sure, I'm confused ;p

Reply:
zzz, nvm me I just read diff below, let me resolve this whole thread and let's pretend none of those comments happend :)

Reply:
We need to convert it to `ts` at some point